plugins/catalog-graph/src/components/CatalogGraphPage/SelectedOwnbyFilter.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { alertApiRef, useApi } from '@backstage/core-plugin-api';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import {
+  Box,
+  Checkbox,
+  FormControlLabel,
+  makeStyles,
+  TextField,
+  Typography,
+} from '@material-ui/core';
+import CheckBoxIcon from '@material-ui/icons/CheckBox';
+import CheckBoxOutlineBlankIcon from '@material-ui/icons/CheckBoxOutlineBlank';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { Autocomplete } from '@material-ui/lab';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import useAsync from 'react-use/lib/useAsync';
+
+const useStyles = makeStyles({
+  formControl: {
+    maxWidth: 300,
+  },
+});
+
+export type Props = {
+  value: string[] | undefined;
+  onChange: (value: string[] | undefined) => void;
+};
+
+export const SelectedOwnByFilter = ({ value, onChange }: Props) => {
+  const classes = useStyles();
+  const alertApi = useApi(alertApiRef);
+  const catalogApi = useApi(catalogApiRef);
+
+  const { error, value: ownedby } = useAsync(async () => {
+    return await catalogApi.getEntityFacets({ facets: ['Group', 'User'] }).then(
```

Comment:
There's no way of getting candidates inside relations using the facets endpoint, sorry. This is because of how it's structured. Best you can do today as far as I can see it, is to do

```ts
getEntities({
  filter: { 'relations.ownedBy': CATALOG_FILTER_EXISTS },
  fields: ['relations']
})
.then(response => response.items.map(stringifyEntitiRef));
```

Then you'll get all the relations of all the entities, but only those that have any ownedBy relation at all. But that's probably gonna be a large request to make in a big catalog.

You could also list all users and groups, which isn't much cheaper

```ts
getEntities({
  filter: { kind: ['Group', 'User'] },
  fields: ['kind', 'metadata.name', 'metadata.namespace'],
})
.then(response => response.items.map(stringifyEntitiRef));
```

Best would be to somehow instead inspect all of the nodes of the current graph if we have those entities available, then we wouldn't have to call out to the catalog at all.

Reply:
makes sense